index.d.ts

```diff
@@ -0,0 +1,16 @@
+import * as DataFactory from '@rdfjs/data-model';
+import { NamedNode } from 'rdf-js';
+
+interface NamespaceBuilder {
+  (property: TemplateStringsArray): NamedNode;
+  (property: string): NamedNode;
+  [property: string]: NamedNode;
+}
```

Comment:
One last change. Because the indexer only implements a getter I suggest that extend a readonly `Record` instead:

```ts
interface NamespaceBuilder extends Readonly<Record<string, NamedNode>> {
  (property: TemplateStringsArray): NamedNode;
  (property: string): NamedNode;
}
```

This will explicitly prevent attempts to do `ns.foo = barNode`